src/blueprints/tv2/factories/tv2-audio-action-factory.ts

```diff
@@ -1,44 +1,116 @@
 import { TransitionType } from '../../../model/enums/transition-type'
-import { DeviceType } from '../../../model/enums/device-type'
 import { PieceActionType } from '../../../model/enums/action-type'
+import { Tv2AudioTimelineObjectFactory } from '../value-objects/factories/tv2-audio-timeline-object-factory'
+import { Tv2BlueprintConfiguration } from '../value-objects/tv2-blueprint-configuration'
+import { Tv2PieceMetadata } from '../value-objects/tv2-metadata'
+
 
 export class Tv2AudioActionFactory {
+  constructor(private readonly audioTimelineObjectFactory: Tv2AudioTimelineObjectFactory) { }
 
-  public createStopAudioBedAction(): PieceAction {
-    const duration: number = 1000
+  public createAudioActions(blueprintConfiguration: Tv2BlueprintConfiguration): Action[] {
+    return [
+      this.createStopAudioBedAction(),
+      this.createMicrophoneUpAction(blueprintConfiguration),
+      this.createMicrophoneDownAction(blueprintConfiguration),
+      this.createResynchronizeAudioAction(),
+      this.createFadePersistedAudioAction()
+    ]
+  }
+
+  private createFadePersistedAudioAction(): PieceAction {
     const pieceInterface: PieceInterface = {
-      id: 'stopAudioBedPiece',
-      name: 'Stop audio bed',
+      ...this.createDefaultAudioPieceInterface(),
+      id: 'fadePersistedAudioPiece',
+      name: 'Fade Persisted Audio',
+      metadata: this.createFadePersistedAudioMetadata()
+    }
+    return {
+      id: 'fadePersistedAudioAction',
+      name: 'Fade Persisted Audio',
+      type: PieceActionType.INSERT_PIECE_AS_ON_AIR,
+      data: pieceInterface
+    }
+  }
+
+  private createFadePersistedAudioMetadata(): Tv2PieceMetadata {
+    return {
+      sisyfosPersistMetaData: {
+        sisyfosLayers: [],
+        acceptsPersistedAudio: false,
+        wantsToPersistAudio: false
+      }
+    }
+  }
+
+  private createMicrophoneUpAction(blueprintConfiguration: Tv2BlueprintConfiguration): PieceAction {
+    const pieceInterface: PieceInterface = {
+      ...this.createDefaultAudioPieceInterface(),
+      id: 'microphoneUpPiece',
+      name: 'Microphone Up',
+      timelineObjects: [
+        this.audioTimelineObjectFactory.createMicrophoneUpTimelineObject(blueprintConfiguration)
+      ]
+    }
+    return {
+      id: 'microphoneUpAction',
+      name: 'Microphone Up',
+      type: PieceActionType.INSERT_PIECE_AS_ON_AIR,
+      data: pieceInterface
+    }
+  }
+
+  private createDefaultAudioPieceInterface(): PieceInterface {
+    return {
+      id: '',
+      name: '',
+      duration: 0,
       partId: '',
       type: PieceType.AUDIO,
-      layer: Tv2SourceLayer.AUDIO_BED,
       pieceLifespan: PieceLifespan.WITHIN_PART,
       transitionType: TransitionType.NO_TRANSITION,
+      layer: Tv2SourceLayer.AUDIO_ACTION_COMMAND,
       isPlanned: false,
       start: 0,
-      duration,
       preRollDuration: 0,
       postRollDuration: 0,
       tags: [],
+      timelineObjects: [],
+    }
+  }
```

Comment:
This. might just be me, but having an empty id, does not seem like a valid declaration. Instead of creating a default/empty piece, you could make a function that creates an AudioPieceInterface given a subset. By doing so, you can require that some of the attributes needs to be set, like `id`. E.g.
```typescript
export class Tv2AudioActionFactory {
  private createAudioPieceInterface(piece: Pick<PieceInterface, 'id'> & Partial<PieceInterface>): PieceInterface {
    // ...
  }
}
```

Reply:
As long as this isn't propagated to the TimelineObjects

Reply:
Have done as suggested by Anders in action factories.